utils/splunk/splunk_telemetry_base.py

```diff
@@ -55,8 +55,10 @@ def check(self, instance):
                 self.service_check(self.SERVICE_CHECK_NAME, AgentCheck.OK)
         except Exception as e:
             self.service_check(self.SERVICE_CHECK_NAME, AgentCheck.CRITICAL, tags=instance.tags, message=str(e))
-            self.log.exception("Splunk event exception: %s" % str(e))
-            raise CheckException("Error getting Splunk data, please check your configuration. Message: " + str(e))
+            if not instance.instance_config.ignore_saved_search_errors:
+                self.log.exception("Splunk event exception: %s" % str(e))
+                raise CheckException("Error getting Splunk data, please check your configuration. Message: " + str(e))
+            self.log.warning("An exception occured in splunk telemetry and ignored as ignore_saved_search_errors flag is true.")
```

Comment:
can we show the actual exception, or at least the message, as well?

Or move line 59, self.log.exception, outside of the if-structure?

Reply:
done.